### PR TITLE
[fingerprint] spawn ExpoConfigLoader with the projectRoot cwd

### PR DESCRIPTION
**Pull request by a-gierczak**
# Why

Using `__dirname` as `cwd` breaks config loading in configs that e.g. rely on `ts-node/register`.
See https://github.com/expo/expo/issues/29793 for more context.

# How

Don't override the `cwd` when spawning the `ExpoConfigLoader.js` node process. It should persist the `cwd` from the parent process.

# Test Plan

1. Clone the reproduction repo: https://github.com/a-gierczak/expo-fingerprint-tsnode-bug-repro
2. Run `yarn install @expo/fingerprint` to install it locally
3. Run `yarn fingerprint > /path/to/repo`
4. It should produce the `Cannot get Expo config from an Expo project` error
5. `yarn link` the local (fixed) version
6. `yarn add @expo/fingerprint` again, to use the locally-linked version
7. Run  `yarn fingerprint > /path/to/repo` again
8. It should show the fingerprint without errors

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/fingerprint/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
 - Fixed `diffFingerprints()` missing some items when the second fingerprint has fewer items than the first fingerprint. ([#29709](https://github.com/expo/expo/pull/29709) by [@kudo](https://github.com/kudo))
 
+- Fixed Expo config sourcer using incorrect `cwd` by ([#29810](https://github.com/expo/expo/pull/29810) by [@a-gierczak](https://github.com/a-gierczak))
+
```
Comment (Kudo) [inline packages/@expo/fingerprint/CHANGELOG.md:16]:
```suggestion
- Fixed Expo config sourcer using incorrect `cwd` by ([#29810](https://github.com/expo/expo/pull/29810) by [@a-gierczak](https://github.com/a-gierczak))
```

Comment (Kudo) [reply packages/@expo/fingerprint/CHANGELOG.md:16]: nit: no newline between changelog subitems.

File: packages/@expo/fingerprint/src/sourcer/Expo.ts [hunk 2]
```diff
@@ -28,11 +28,11 @@ export async function getExpoConfigSourcesAsync(
...
-    const { stdout } = await spawnAsync(
-      'node',
-      [getExpoConfigLoaderPath(), path.resolve(projectRoot), ignoredFile],
-      { cwd: __dirname }
-    );
+    const { stdout } = await spawnAsync('node', [
+      getExpoConfigLoaderPath(),
+      path.resolve(projectRoot),
+      ignoredFile,
+    ]);
```
Comment (Kudo) [inline packages/@expo/fingerprint/src/sourcer/Expo.ts:35]:
```suggestion
    const { stdout } = await spawnAsync(
      'node',
      [getExpoConfigLoaderPath(), path.resolve(projectRoot), ignoredFile],
      { cwd: projectRoot }
    );
```

nit: using `projectRoot` probably may be explicit and clear?

**Review by Kudo — COMMENTED** [review]
thanks for having this pr. just leave some nit comments and hope that makes sense to you

File: packages/@expo/fingerprint/src/sourcer/Expo.ts (hunk 2 above)
Comment (a-gierczak) [reply packages/@expo/fingerprint/src/sourcer/Expo.ts]: makes sense, thx 👍

**Review by Kudo — APPROVED** [review]
thanks again for helping the fix!

**Pull request merged**